.github/workflows/regression.yml

```diff
@@ -0,0 +1,89 @@
+name: Regression Testing
+
+on:
+  push:
+    branches:
+      - 'develop'
+      - 'master'
+  pull_request:
+    branches: 
+      - 'develop'
+      - 'master'
+
+jobs:
+  build:
+    name: Build SU2
+    strategy:
+      fail-fast: false
+      matrix: 
+        config_set: [BaseMPI, ReverseMPI, ForwardMPI, BaseNoMPI, ReverseNoMPI, ForwardNoMPI]
+        include:
+          - config_set: BaseMPI
+            flags: '-Denable-pywrapper=true'
+          - config_set: ReverseMPI
+            flags: '-Denable-autodiff=true -Denable-normal=false -Denable-pywrapper=true'
+          - config_set: ForwardMPI
+            flags: '-Denable-directdiff=true -Denable-normal=false'          
+          - config_set: BaseNoMPI
+            flags: '-Denable-pywrapper=true -Dwith-mpi=disabled'
+          - config_set: ReverseNoMPI
+            flags: '-Denable-autodiff=true -Denable-normal=false -Dwith-mpi=disabled -Denable-pywrapper=true'
+          - config_set: ForwardNoMPI
+            flags: '-Denable-directdiff=true -Denable-normal=false -Dwith-mpi=disabled'  
+    runs-on: ubuntu-latest
+    steps:
+      - name: Dump matrix context
+        env:
+            MATRIX_CONTEXT: ${{ toJson(matrix) }}
+        run: echo "$MATRIX_CONTEXT"
+      - name: Build 
+        uses: docker://su2code/build-su2:20191029
```

Comment:
What is the significance of pulling from the docker with the date 20191029? Any issue there, or will it always be the latest (and just a naming thing)?

This appears again below.

Reply:
Putting there a specific tag makes sure that we don't accidentally break the regression tests, because anything without a specific tag will be tagged as "latest" once you push it to the docker repository.

Reply:
So it just refers to a specific "named" version of the image (i thought the date would make sense).